tests/api.c

```diff
@@ -6863,6 +6871,1086 @@ static int test_wc_Arc4Process (void)
+
+    DECLARE_VAR_INIT(in, byte, inLen, inStr, NULL);
+    DECLARE_VAR(plain, byte, 25, NULL);
+    DECLARE_VAR(cipher, byte, 128, NULL);
+
+    /* Initialize stack structures. */
+    XMEMSET(&rng, 0, sizeof(rng));
+    XMEMSET(&key, 0, sizeof(key));
+
+    ret = wc_InitRsaKey_ex(&key, NULL, INVALID_DEVID);
+    if (ret == 0) {
+        ret = wc_InitRng(&rng);
+    }
+    if (ret == 0) {
+        ret = wc_MakeRsaKey(&key, 1024, 65537, &rng);
+    }
+    /* Encrypt */
+    printf(testingFmt, "wc_RsaPublicEncrypt_ex()");
+    if (ret == 0) {
+        ret = wc_RsaPublicEncrypt_ex(in, inLen, cipher, cipherSz, &key, &rng,
+                WC_RSA_OAEP_PAD, WC_HASH_TYPE_SHA, WC_MGF1SHA1, NULL, 0);
+        if (ret >= 0) {
+            idx = ret;
+            ret = 0;
+        } else {
+            ret = SSL_FATAL_ERROR;
+        }
+    }
+
+    /*Pass bad args.*/
+   /* Tests RsaPublicEncryptEx again. No need duplicate. */
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+    /* Decrypt */
+    printf(testingFmt, "wc_RsaPrivateDecrypt_ex()");
+    if (ret == 0) {
+        ret = wc_RsaSetRNG(&key, &rng);
+        if (ret == 0) {
+            ret = wc_RsaPrivateDecrypt_ex(cipher, (word32)idx,
+                    plain, plainSz, &key, WC_RSA_OAEP_PAD, WC_HASH_TYPE_SHA,
+                    WC_MGF1SHA1, NULL, 0);
+        }
+        if (ret >= 0) {
+            if (!XMEMCMP(plain, inStr, 25)) {
+                ret = 0;
+            } else {
+                ret = SSL_FATAL_ERROR;
+            }
+        }
+    }
+
+    /*Pass bad args.*/
+   /* Tests RsaPrivateDecryptEx() again. No need duplicate. */
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+    printf(testingFmt, "wc_RsaPrivateDecryptInline_ex()");
+    if (ret == 0) {
+        ret = wc_RsaPrivateDecryptInline_ex(cipher, (word32)idx,
+                &res, &key, WC_RSA_OAEP_PAD, WC_HASH_TYPE_SHA,
+                WC_MGF1SHA1, NULL, 0);
+
+        if (ret >= 0) {
+            if (!XMEMCMP(inStr, res, 25)) {
+                ret = 0;
+            } else {
+                ret = SSL_FATAL_ERROR;
+            }
+        }
+    }
+
+    FREE_VAR(in, NULL);
+    FREE_VAR(plain, NULL);
+    FREE_VAR(cipher, NULL);
+    if (wc_FreeRsaKey(&key) || ret != 0) {
+        ret = SSL_FATAL_ERROR;
+    }
+    if (wc_FreeRng(&rng) || ret != 0) {
+        ret = SSL_FATAL_ERROR;
+    }
+
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+#endif
+    return 0;
+
+} /* END test_wc_RsaPublicEncryptDecrypt_ex */
+
+/*
+ * Tesing wc_RsaSSL_Sign() and wc_RsaSSL_Verify()
+ */
+static int test_wc_RsaSSL_SignVerify (void)
+{
+#if !defined(NO_RSA) && defined(WOLFSSL_KEY_GEN)
+    RsaKey  key;
+    WC_RNG  rng;
+    const char* inStr = "Everyone gets Friday off.";
+    const word32 outSz = 128;
+    const word32 plainSz = 25;
+    word32  inLen = (word32)XSTRLEN(inStr);
+    word32  idx = 0;
+    int     ret;
+
+    DECLARE_VAR_INIT(in, byte, inLen, inStr, NULL);
+    DECLARE_VAR(out, byte, 128, NULL);
```

Comment:
Please apply plainSz and outSz where applicable in this function instead of 25 and 128.